### ISSUE TITLE
Safe Methods vs Redirection (Trac #10)

### ISSUE THREAD

**Issue by mnot** [httpwg/httpbis-issues#10]
```text
#!html
           <p>
                Section <a href='http://www.apps.ietf.org/rfc/rfc2616.html#sec-10.3.2'>10.3.2</a> (301 Moved Permanently) contains the paragraph
            </p>
<pre class='error'>
    If the 301 status code is received in response to a request <span class='diff'>other
        than GET or HEAD</span>, the user agent MUST NOT automatically redirect the
    request unless it can be confirmed by the user, since this might
    change the conditions under which the request was issued.</pre>
            <p>
                which fails to consider that there are many other request methods
                that are safe to automatically redirect, and further that the user agent
                is able to make that determination based on the request method semantics.
                In particular, the OPTIONS method is always safe to automatically redirect.
                Unfortunately, the paragraph was written long before there was OPTIONS,
                and was never updated to reflect the extensibility of methods.  The
                same problem paragraph is found in sections <a href='http://www.apps.ietf.org/rfc/rfc2616.html#sec-10.3.3'>10.3.3</a> and <a href='http://www.apps.ietf.org/rfc/rfc2616.html#sec-10.3.8'>10.3.8</a>.
            </p>
            <p>
                The above should be replaced with
            </p>
<pre class='correct'>
    If the 301 status code is received in response to a request <span class='diff'>method
        that is known to be "safe", as defined in section 9.1.1, then the
        request MAY be automatically redirected by the user agent without
        confirmation.  Otherwise</span>, the user agent MUST NOT automatically
    redirect the request unless it is confirmed by the user, since the
    new URI might change the conditions under which the request was issued.</pre>
            <p>
                along with similar changes for sections <a href='http://www.apps.ietf.org/rfc/rfc2616.html#sec-10.3.3'>10.3.3</a> and <a href='http://www.apps.ietf.org/rfc/rfc2616.html#sec-10.3.8'>10.3.8</a>.
                It would also be helpful for each of the method definition sections
                to specifically define whether or not the method is safe.
                OPTIONS, GET, and HEAD are all safe in RFC 2616.
                HTTP extensions like WebDAV define additional safe methods.
            </p>
            

```

Migrated from https://trac.ietf.org/ticket/10
```json
{
    "status": "closed",
    "changetime": "2010-03-17T10:00:56",
    "_ts": "1268820056000000",
    "description": "{{{\n#!html\n           <p>\n                Section <a href='http://www.apps.ietf.org/rfc/rfc2616.html#sec-10.3.2'>10.3.2</a> (301 Moved Permanently) contains the paragraph\n            </p>\n<pre class='error'>\n    If the 301 status code is received in response to a request <span class='diff'>other\n        than GET or HEAD</span>, the user agent MUST NOT automatically redirect the\n    request unless it can be confirmed by the user, since this might\n    change the conditions under which the request was issued.</pre>\n            <p>\n                which fails to consider that there are many other request methods\n                that are safe to automatically redirect, and further that the user agent\n                is able to make that determination based on the request method semantics.\n                In particular, the OPTIONS method is always safe to automatically redirect.\n                Unfortunately, the paragraph was written long before there was OPTIONS,\n                and was never updated to reflect the extensibility of methods.  The\n                same problem paragraph is found in sections <a href='http://www.apps.ietf.org/rfc/rfc2616.html#sec-10.3.3'>10.3.3</a> and <a href='http://www.apps.ietf.org/rfc/rfc2616.html#sec-10.3.8'>10.3.8</a>.\n            </p>\n            <p>\n                The above should be replaced with\n            </p>\n<pre class='correct'>\n    If the 301 status code is received in response to a request <span class='diff'>method\n        that is known to be \"safe\", as defined in section 9.1.1, then the\n        request MAY be automatically redirected by the user agent without\n        confirmation.  Otherwise</span>, the user agent MUST NOT automatically\n    redirect the request unless it is confirmed by the user, since the\n    new URI might change the conditions under which the request was issued.</pre>\n            <p>\n                along with similar changes for sections <a href='http://www.apps.ietf.org/rfc/rfc2616.html#sec-10.3.3'>10.3.3</a> and <a href='http://www.apps.ietf.org/rfc/rfc2616.html#sec-10.3.8'>10.3.8</a>.\n                It would also be helpful for each of the method definition sections\n                to specifically define whether or not the method is safe.\n                OPTIONS, GET, and HEAD are all safe in RFC 2616.\n                HTTP extensions like WebDAV define additional safe methods.\n            </p>\n            \n\n}}}",
    "reporter": "mnot@pobox.com",
    "cc": "",
    "resolution": "fixed",
    "time": "2007-12-12T09:38:25",
    "component": "p2-semantics",
    "summary": "Safe Methods vs Redirection",
    "priority": "urgent",
    "keywords": "",
    "milestone": "09",
    "owner": "",
    "type": "design",
    "severity": "Active WG Document"
}
```

**Comment by mnot** [comment]
Trac update at 20071222T00:17:38: 
* fielding@gbiv.com commented:

> Fixed in [88]


* fielding@gbiv.com changed milestone from "" to "01"
* fielding@gbiv.com changed origin from "" to ""
* fielding@gbiv.com changed resolution from "" to "fixed"
* fielding@gbiv.com changed status from "new" to "closed"
* fielding@gbiv.com changed version from "" to "00"

**Comment by mnot** [comment]
Trac update at 20071222T04:12:18: mnot@pobox.com changed version from "00" to "d00"

**Comment by mnot** [comment]
Trac update at 20080809T12:56:47: julian.reschke@gmx.de changed component from "" to "p2-semantics"

**Comment by mnot** [comment]
Trac update at 20080809T18:52:34: julian.reschke@gmx.de changed origin from "" to "http://lists.w3.org/Archives/Public/ietf-http-wg-old/2001JanApr/0031.html"

**Comment by mnot** [comment]
Trac update at 20100212T10:48:49: 
* julian.reschke@gmx.de commented:

> Maciej Stachowiak points out that we did not fix this everywhere
> (http://lists.w3.org/Archives/Public/ietf-http-wg/2010JanMar/0170.html), the
> introduction to 3xx still says:
> 
> "The action required MAY be carried out by the user agent without interaction
> with the user if and only if the method used in the second request is GET or
> HEAD."
> 


* julian.reschke@gmx.de changed milestone from "01" to "09"
* julian.reschke@gmx.de changed priority from "" to "urgent"
* julian.reschke@gmx.de changed resolution from "fixed" to ""
* julian.reschke@gmx.de changed severity from "" to "Active WG Document"
* julian.reschke@gmx.de changed status from "closed" to "reopened"

**Comment by mnot** [comment]
Trac update at 20100212T11:09:43: julian.reschke@gmx.de commented:

> From [759]:
> 
> Replace "GET or HEAD" by "safe method" in intro to 3xx codes (see #10)
>

**Comment by mnot** [comment]
Trac update at 20100212T16:31:31: julian.reschke@gmx.de commented:

> From [760]:
> 
> Replace "GET or HEAD" by "safe method" in intro to 3xx codes (fix typo) (see #10)
>

**Comment by mnot** [comment]
Trac update at 20100317T10:00:56: 
* mnot@pobox.com changed resolution from "" to "fixed"
* mnot@pobox.com changed status from "reopened" to "closed"
